networking/eth2/src/integration-test/java/tech/pegasys/teku/networking/eth2/BlobsSidecarsByRangeIntegrationTest.java

```diff
@@ -71,6 +71,24 @@ public void shouldReturnAvailableBlobsSideCars() throws Exception {
         .isEqualTo(blockAndState3.getBlock().getParentRoot());
   }
 
+  @Test
+  public void shouldHandleRequestSpanningTheEip4844ForkTransition()
+      throws ExecutionException, InterruptedException, TimeoutException {
+    final Eth2Peer peer =
+        createPeer(TestSpecFactory.createMinimalWithEip4844ForkEpoch(UInt64.valueOf(1)));
+
+    // slot 8 - 13 is in EIP-4844 milestone
+    final SignedBlockAndState latestBlockAndState =
+        peerStorage.chainUpdater().advanceChainUntil(13);
+    peerStorage.chainUpdater().updateBestBlock(latestBlockAndState);
+
+    final List<BlobsSidecar> response = requestBlobsSideCars(peer, UInt64.ONE, UInt64.valueOf(13));
+
+    assertThat(response).hasSize(6);
```

Comment:
Could we also have a test where we check the request? Maybe `Eth2PeerTest` with delegate's `sendRequest` verification. I'd like to see that 
1. nothing is sent when we don't have blobs in range
2. all needed blobs are requested when spanning, especially on borderline

Reply:
Yes good point, will add a test